pkg/controller/networkpolicy/validate.go

```diff
@@ -615,6 +624,71 @@ func (v *antreaPolicyValidator) validateTierForPassAction(tier string, ingress,
 	return "", true
 }
 
+func (v *antreaPolicyValidator) validateEgressMulticastAddress(egressRule []crdv1alpha1.Rule) (string, bool) {
+	for _, r := range egressRule {
+		multicast := false
+		unicast := false
+		otherSelectors := false
+		for _, to := range r.To {
+			if to.IPBlock == nil {
+				continue
+			}
+			toIPAddr, _, err := net.ParseCIDR(to.IPBlock.CIDR)
+			if err != nil {
+				return fmt.Sprintf("invalid multicast groupAddress address (to.IPBlock.CIDR): %v", err.Error()), false
+			}
+			if toIPAddr.IsMulticast() {
+				multicast = true
+			} else {
+				unicast = true
+			}
+			if to.PodSelector != nil || to.NamespaceSelector != nil || to.Namespaces != nil ||
+				to.ExternalEntitySelector != nil || to.ServiceAccount != nil || to.NodeSelector != nil {
+				otherSelectors = true
+			}
+			if multicast && (*r.Action == crdv1alpha1.RuleActionPass || *r.Action == crdv1alpha1.RuleActionReject) {
+				return fmt.Sprintf("multicast does not support action Pass or Reject"), false
+			}
+		}
+		if multicast && unicast {
+			return fmt.Sprintf("can not set multicast groupAddress and unicast ip address at the same time"), false
+		}
+		if multicast && otherSelectors {
+			return fmt.Sprintf("can not set multicast groupAddress and selectors at the same time"), false
+		}
+	}
+	return "", true
+}
+
+func validateIGMPProtocol(protocol crdv1alpha1.NetworkPolicyProtocol) (string, bool) {
+	if protocol.IGMP.GroupAddress == "" {
+		return "", true
+	}
+	groupIP := net.ParseIP(protocol.IGMP.GroupAddress)
+	if !groupIP.IsMulticast() {
+		return fmt.Sprintf("groupAddress %+v is not multicast address", groupIP), false
+	}
+
+	return "", true
+}
+
+func (v *antreaPolicyValidator) validateMulticastIGMP(ingressRules, egressRules []crdv1alpha1.Rule) (string, bool) {
+	for _, r := range append(ingressRules, egressRules...) {
+		for _, protocol := range r.Protocols {
+			if protocol.IGMP != nil {
```

Comment:
I remember you said a rule's protocols must be all IGMP or none and it's checked by webhook, and many of your code relies on the assumption, but I don't find such validation here.
What if users create below rules? I think they will not work as expected.
```
- egress:
  - protocols:
     - icmp: {}
     - igmp: {}
```

Reply:
done

Reply:
I was just giving an example, it can't just check ICMP and IGMP. There are other cases like
```
- egress:
  - ports:
    - protocol: TCP
      port: 80
    protocols:
      igmp: {}
```